airbyte-integrations/connectors/source-zuora/source_zuora/zuora_client.py

```diff
@@ -0,0 +1,275 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+import json as j
+from datetime import datetime
+from math import ceil
+from typing import Dict, Iterable, List
+
+import pendulum
+import requests
+from airbyte_cdk import AirbyteLogger
+
+from .zuora_auth import ZuoraAuthenticator
+from .zuora_errors import EndDateError, ZOQLQueryCannotProcessObject, ZOQLQueryFailed, ZOQLQueryFieldCannotResolve, ZOQLQueryNotValid
+
+
+class ZoqlExportClient:
+
+    """
+    # TODO: Add the description about the ZOQL EXPORT + Links
+    # TODO: Add description to class methods
+    """
+
+    logger = AirbyteLogger()
+
+    def __init__(self, authenticator: Dict, start_date: str, window_in_days: int, client_id: str, client_secret: str, is_sandbox: bool):
+        self.authenticator = authenticator
+        self.start_date = start_date
+        self.window_in_days = window_in_days
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.is_sandbox = is_sandbox
+        self.retry_max = 3
+
+    # Define base url from is_sandbox arg.
+    @property
+    def url_base(self) -> str:
+        if self.is_sandbox:
+            return "https://rest.apisandbox.zuora.com"
+        else:
+            return "https://rest.zuora.com"
```

Comment:
Is it possible to reuse it from `ZuoraAuthenticator.endpoint` to reduce duplication?

Reply:
Done